### ISSUE TITLE
refactor json schema generation

### ISSUE THREAD

**Issue by azahnen** [interactive-instruments/ldproxy#165]
Depends on #162.

Two topics should be addressed:

### Transformations

Schema transformations, e.g. flattening of objects, should be the responsibility of the provider. Such transformations should be defined either in the provider configuration or for a specific output format in the service configuration. A consumer should then be able to get the transformed schema from the provider.

### Encoding

Schemas might be encoded in different output formats, just like features. Therefore we should implement a similar mechanism to register encodings with a mime type.

**Comment by azahnen** [comment]
The first topic was addressed in #483. I only refactored the existing functionality, which means that currently only the flatten transformation is considered. As a follow-up, the `WithTransformationsApplied` visitor should apply all given transformations to the `FeatureSchema`. I would also like to move some classes to xtraplatform or to more fitting ogcapi modules.

I think we should ignore the second topic for now. It would only apply to the schema endpoint, which currently only has one encoding anyways. But schemas are also used as part of OpenApi documents, which makes the topic much more complex than expected.
